restli-server/src/main/java/com/linkedin/restli/internal/server/methods/MethodAdapterRegistry.java

```diff
@@ -1,130 +1,23 @@
-  {
-    _adapters = buildAdapterRegistry();
-    _responseBuilders = buildResponseBuilders(errorResponseBuilder);
-  }
-
-  private Map<ResourceMethod, RestLiArgumentBuilder> buildAdapterRegistry()
-  {
-    Map<ResourceMethod, RestLiArgumentBuilder> result =
-        new HashMap<>(ResourceMethod.values().length);
-    result.put(ResourceMethod.GET, new GetArgumentBuilder());
-    result.put(ResourceMethod.BATCH_GET, new BatchGetArgumentBuilder());
-    result.put(ResourceMethod.FINDER, new CollectionArgumentBuilder());
-    result.put(ResourceMethod.BATCH_FINDER, new CollectionArgumentBuilder());
-    result.put(ResourceMethod.CREATE, new CreateArgumentBuilder());
-    result.put(ResourceMethod.PARTIAL_UPDATE, new PatchArgumentBuilder());
-    result.put(ResourceMethod.UPDATE, new UpdateArgumentBuilder());
-    result.put(ResourceMethod.DELETE, new GetArgumentBuilder());
-    result.put(ResourceMethod.ACTION, new ActionArgumentBuilder());
-    result.put(ResourceMethod.BATCH_UPDATE, new BatchUpdateArgumentBuilder());
-    result.put(ResourceMethod.BATCH_PARTIAL_UPDATE, new BatchPatchArgumentBuilder());
-    result.put(ResourceMethod.BATCH_CREATE, new BatchCreateArgumentBuilder());
-    result.put(ResourceMethod.BATCH_DELETE, new BatchDeleteArgumentBuilder());
-    result.put(ResourceMethod.GET_ALL, new CollectionArgumentBuilder());
-    return Collections.unmodifiableMap(result);
-  }
-
-  private Map<ResourceMethod, RestLiResponseBuilder<?>> buildResponseBuilders(ErrorResponseBuilder errorResponseBuilder)
-  {
-    Map<ResourceMethod, RestLiResponseBuilder<?>> result =
-        new HashMap<>(ResourceMethod.values().length);
-
-    result.put(ResourceMethod.GET, new GetResponseBuilder());
-    result.put(ResourceMethod.BATCH_GET, new BatchGetResponseBuilder(errorResponseBuilder));
-    result.put(ResourceMethod.FINDER, new FinderResponseBuilder());
-    result.put(ResourceMethod.CREATE, new CreateResponseBuilder());
-    result.put(ResourceMethod.PARTIAL_UPDATE, new PartialUpdateResponseBuilder());
-    result.put(ResourceMethod.UPDATE, new UpdateResponseBuilder());
-    result.put(ResourceMethod.DELETE, new DeleteResponseBuilder());
-    result.put(ResourceMethod.ACTION, new ActionResponseBuilder());
-    result.put(ResourceMethod.BATCH_UPDATE, new BatchUpdateResponseBuilder(errorResponseBuilder));
-    result.put(ResourceMethod.BATCH_PARTIAL_UPDATE, new BatchPartialUpdateResponseBuilder(errorResponseBuilder));
-    result.put(ResourceMethod.BATCH_CREATE, new BatchCreateResponseBuilder(errorResponseBuilder));
-    result.put(ResourceMethod.BATCH_DELETE, new BatchDeleteResponseBuilder(errorResponseBuilder));
-    result.put(ResourceMethod.BATCH_FINDER, new BatchFinderResponseBuilder(errorResponseBuilder));
-    result.put(ResourceMethod.GET_ALL, new GetAllResponseBuilder());
-
-    return Collections.unmodifiableMap(result);
-  }
 
+public interface MethodAdapterRegistry {
```

Comment:
Simply changing this class to an interface is backward-incompatible. There may be places where this is constructed explicitly. I suggest keeping `MethodAdapterRegistry`, deprecating it (with a javadoc saying to use `DefaultMethodAdapterRegistry`), and having its implementation defer to the `DefaultMethodAdapterRegistry` (perhaps by extending it). This means we will need to introduce the interface as a new file with a unique name.

Since I don't want something like `MethodAdapterRegistryInterface`, I suggest we consider renaming this altogether. I propose using the naming convention `MethodBuildersRegistry`, `DefaultMethodBuildersRegistry`, and the now-deprecated class `MethodAdapterRegistry` will extend `DefaultMethodBuildersRegistry` (so that it relies on its functionality).

The exact naming can be adjusted, but my core idea is to use new naming.